`terminal/tui_spec.lua` functional tests fail under tmux.

### Problem

I spent some time this morning working to identify why 2 functional tests were unexpectedly failing on latest master when running them locally. I went through the typical debugging steps, did a make distclean, ran the functional tests in an ubuntu 20.04 docker container, then finally tried running the tests under xterm without tmux which ended up fixing the issue and everything succeeded.

I then tried a few other terminals with and without tmux. Kitty passes all functional tests without tmux, then with tmux the same two tests fail, I did the same with xterm with and without and got the same result, and finally I tried attaching to the same docker container under tmux and then not under tmux and had the same result even though the `$TERM` variable was xterm regardless of where I attached from.

I also tried changing setting `$TERM` to `xterm`, and erasing `$TERM_PROGRAM` and `$TERM_PROGRAM_VERSION` and the issue still occurred under tmux.

My guess for the cause is that during the tests nvim tries querying the terminal for support of a certain feature via an escape code through stdout, and nvim doesn't support it? I'm unsure if that is possible though since the nvim session isn't really attached to stdout and stdin during functional tests.

This also occurs when running tmux without my config via:

```
tmux -L test -f /dev/null
```

Not a huge deal but it might save other newcomers to nvim development some headache to skip these tests under tmux if thats possible, I'm happy to do that unless this is an actual issue that should be fixed.

The test results summary when running under tmux is below:
```
-------- Global test environment teardown.
======== 328 tests from 15 test files ran. (26538.00 ms total)
PASSED   325 tests.
SKIPPED  1 test, listed below:
SKIPPED  test/functional/terminal/tui_spec.lua @ 3105: TUI 't_Co' (terminal colors) TERM=interix uses 8 colors
FAILED   2 tests, listed below:
FAILED   test/functional/terminal/tui_spec.lua @ 1762: TUI forwards :term palette colors with termguicolors
test/functional/terminal/tui_spec.lua:1824: Row 5 did not match.
Expected:
  |^tty ready                                         |
  |{4:text}colortext                                     |
  |                                                  |
  |                                                  |
  |*{6:^^^^^^^}{7:                                           }|
  |:set notermguicolors                              |
  |{3:-- TERMINAL --}                                    |
Actual:
  |^tty ready                                         |
  |{4:text}colortext                                     |
  |                                                  |
  |                                                  |
  |*{6:^^^^^^^                                           }|
  |:set notermguicolors                              |
  |{3:-- TERMINAL --}                                    |

To print the expect() call that would assert the current screen state, use
screen:snapshot_util(). In case of non-deterministic failures, use
screen:redraw_debug() to show all intermediate screen states.

Snapshot:
screen:expect([[
  ^tty ready                                         |
  {4:text}colortext                                     |
                                                    |*2
  {6:^^^^^^^                                           }|
  :set notermguicolors                              |
  {3:-- TERMINAL --}                                    |
]])

stack traceback:
        test/functional/ui/screen.lua:852: in function '_wait'
        test/functional/ui/screen.lua:517: in function 'expect'
        test/functional/terminal/tui_spec.lua:1824: in function <test/functional/terminal/tui_spec.lua:1762>

FAILED   test/functional/terminal/tui_spec.lua @ 1837: TUI supports undercurl and underdouble when run in :terminal
test/functional/terminal/tui_spec.lua:1848: Row 1 did not match.
Expected:
  |*{5:fooba}^r                                            |
  |{4:~                                                 }|
  |{4:~                                                 }|
  |{4:~                                                 }|
  |{2:[No Name] [+]                                     }|
  |{3:-- VISUAL LINE --}                                 |
  |{3:-- TERMINAL --}                                    |
Actual:
  |*{UNEXPECTED underline = true:fooba}^r                                            |
  |{4:~                                                 }|
  |{4:~                                                 }|
  |{4:~                                                 }|
  |{2:[No Name] [+]                                     }|
  |{3:-- VISUAL LINE --}                                 |
  |{3:-- TERMINAL --}                                    |

To print the expect() call that would assert the current screen state, use
screen:snapshot_util(). In case of non-deterministic failures, use
screen:redraw_debug() to show all intermediate screen states.

Snapshot:
screen:set_default_attr_ids({
  [1] = { reverse = true },
  [2] = { reverse = true, bold = true },
  [3] = { bold = true },
  [4] = { foreground = 12 },
  [5] = { undercurl = true },
  [6] = { underdouble = true },
  [7] = { underline = true },
})

screen:expect([[
  {7:fooba}^r                                            |
  {4:~                                                 }|*3
  {2:[No Name] [+]                                     }|
  {3:-- VISUAL LINE --}                                 |
  {3:-- TERMINAL --}                                    |
]])

stack traceback:
        test/functional/ui/screen.lua:852: in function '_wait'
        test/functional/ui/screen.lua:517: in function 'expect'
        test/functional/terminal/tui_spec.lua:1848: in function <test/functional/terminal/tui_spec.lua:1837>


 1 SKIPPED TEST
 2 FAILED TESTS
------------------------------------------------------------------------------
$NVIM_LOG_FILE: /home/gabe/Code/neovim/build/.nvimlog
(last 10 lines)
DBG 2025-05-25T14:58:36.165 T327.1574556.0 inbuf_poll:516: blocking... events=false
DBG 2025-05-25T14:58:36.165 T327.1574556.0 inbuf_poll:516: blocking... events=true
DBG 2025-05-25T14:58:36.165 T327.1574556.0 receive_msgpack:210: ch 1: parsing 22 bytes from msgpack Stream: 0x618122071860
DBG 2025-05-25T14:58:36.165 T327.1574556.0 RPC: <- 1: [request]   id=22: nvim_eval
DBG 2025-05-25T14:58:36.166 T327.1574556.0 handle_request:347: RPC: scheduled nvim_eval
DBG 2025-05-25T14:58:36.166 T327.1574556.0 state_enter:99: input: K_EVENT
DBG 2025-05-25T14:58:36.166 T327.1574556.0 handle_nvim_eval:9690: RPC: ch 1: invoke nvim_eval
DBG 2025-05-25T14:58:36.166 T327.1574556.0 RPC: -> 1: [response]  id=22
DBG 2025-05-25T14:58:36.166 T327.1574556.0 inbuf_poll:516: blocking... events=false
DBG 2025-05-25T14:58:36.166 T327.1574556.0 inbuf_poll:516: blocking... events=true
------------------------------------------------------------------------------
E5113: Lua chunk:
-- Tests exited non-zero: 1
CMake Error at /home/gabe/Code/neovim/cmake/RunTests.cmake:98 (message):
  functional tests failed with error: 1


FAILED: test/CMakeFiles/functionaltest /home/gabe/Code/neovim/build/test/CMakeFiles/functionaltest
cd /home/gabe/Code/neovim/build/test && /usr/bin/cmake -D TEST_TYPE=functional -D BUILD_DIR=/home/gabe/Code/neovim/build -D CIRRUS_CI= -D CI_BUILD=OFF -D DEPS_INSTALL_DIR=/home/gabe/Code/neovim/build/usr -D NVIM_PRG=/home/gabe/Code/neovim/build/bin/nvim -D TEST_DIR=/home/gabe/Code/neovim/test -D WORKING_DIR=/home/gabe/Code/neovim -P /home/gabe/Code/neovim/cmake/RunTests.cmake
ninja: build stopped: subcommand failed.
make: *** [Makefile:142: functionaltest] Error 1

```

### Steps to reproduce

```
tmux -L /tmp/clean_tmux_server -f /dev/null
TEST_FILE="test/functional/terminal/" make functionaltest 
```

### Expected behavior

I would expect these tests to skip or pass under tmux.

### Nvim version (nvim -v)

NVIM v0.12.0-dev-434+g8b9500c886 -- Build type: Debug -- LuaJIT 2.1.1744317938

### Vim (not Nvim) behaves the same?

These tests don't exist in vim.

### Operating system/version

OS: Arch Linux x86_64 -- Kernel: 6.14.6-arch1-1

### Terminal name/version

kitty 0.42.0

### $TERM environment variable

tmux-256color

### Installation

latest master -- build from repo

--- a/cmake/RunTests.cmake
+++ b/cmake/RunTests.cmake
@@ -7,6 +7,7 @@ set(ENV{XDG_DATA_HOME} ${BUILD_DIR}/Xtest_xdg/share)
 set(ENV{XDG_STATE_HOME} ${BUILD_DIR}/Xtest_xdg/state)
 unset(ENV{XDG_DATA_DIRS})
 unset(ENV{NVIM})  # Clear $NVIM in case tests are running from Nvim. #11009
+unset(ENV{TMUX})  # Nvim TUI shouldn't think it's running in tmux. #34173
 
 # TODO(dundargoc): The CIRRUS_CI environment variable isn't passed to here from
 # the main CMakeLists.txt, so we have to manually pass it to this script and

